contracts/mesh-provider/src/contract.rs

```diff
@@ -192,16 +194,6 @@ pub fn execute_unstake(
     STAKED.save(deps.storage, (&info.sender, &validator), &stake)?;
     VALIDATORS.save(deps.storage, &validator, &val)?;
 
-    // create a future claim on number of shares (so we can adjust for later slashing)
```

Comment:
Mental model from Sunny: "The provider chain should be resilient in face of Byzantine action from the consumer chain".

This means, the *provider* enforces the unstaking/unbonding period, and only slashes if it receives proof. It doesn't rely on the consumer chain to release tokens or slash arbitrarily.

Following this model, we should always unstake on the provider. We just want to ensure it makes it to the consumer if the consumer if honest. Staking on the other hand, should only happen if the consumer accepts it.

(Basically, we only lock up your tokens if the consumer promises to honor them. But we unlock them according to our local rules, just informing the consumer of the action).